File: js/stack/844backspaceCompare.js
/**
 * @param {string} S
 * @param {string} T
 * @return {boolean}
 */
var backspaceCompare = function(S, T) {
    return getStr(S) === getStr(T);
   function getStr(str){
       let stack = [];
       for (let a of str){
           if(a === '#'){
               stack.pop();
           }else{
               stack.push(a)
           }
       }
       console.log(stack.join(''));
       return stack.join('');
   }
};

let testcase = backspaceCompare("xywrrmp","xywrrmu#p");
console.log(testcase);
export default backspaceCompare;

File: py/stack/844backspaceCompare.py
class Solution:
    def backspaceCompare(self, S: str, T: str) -> bool:
        def getStr(sss: str):
            stack = []
            for s in sss:
                if s == '#':
                    if len(stack) > 0:
                        stack.pop()
                else:
                    stack.append(s)
            return ''.join(stack)

        return getStr(S) == getStr(T)

solution = Solution()
ans = solution.backspaceCompare("y#fo##f","y#f#o##f")
print(ans)